### PR TITLE
Add suite of unit tests for WhiteNoiseKernel and ZeroLazyVariable

### PR DESCRIPTION
**Pull request by jacobrgardner**
This also fixes some issues that turned up with them in writing the unit tests for the batch setting. 

This PR begins work on #224. I think it is best to make PRs for unit tests as I make them rather than trying to do a big push for them all in one PR, since I may also be able to fix minor issues as I make them as in this PR.

cc/ @gpleiss @Balandat

### CODE REVIEW

**Comment by jacobrgardner** [comment]
Actually, this WhiteNoiseKernel will still do the wrong thing in the multitask setting, which is what I'm doing unit tests for now. 

I am going to propose that we start enforcing only 3 dimensional ever means batch mode.

File: gpytorch/kernels/white_noise_kernel.py [hunk 1]
```diff
@@ -13,16 +13,24 @@ def __init__(self, variances):
         super(WhiteNoiseKernel, self).__init__()
         self.register_buffer("variances", variances)
 
+        # Enforce that variances are batch_size x data_size x task_size
```
Comment (jacobrgardner) [inline gpytorch/kernels/white_noise_kernel.py]: @Balandat @gpleiss What do you guys think of this pattern? For any parameter we define internally, we always reshape it to have 3 dimensions that denote `batch_size x data_size x task_size`. If a user initializes with a 3 dimensional tensor, we keep it as is. If they pass in a 2 dimensional tensor, we assume it is `data_size x task_size` and unsqueeze a singleton batch dimension. If they pass in a  1 dimensional tensor, we assume it is `data_size` and unsqueeze it to `1 x data_size x 1`.

Comment (Balandat) [reply gpytorch/kernels/white_noise_kernel.py]: Yeah I like this pattern, I think I had suggested that in an earlier discussion.

Comment (jacobrgardner) [reply gpytorch/kernels/white_noise_kernel.py]: Great. In that case, I think these are all ready to go. I'll work on some multitask unit tests next.

**Pull request merged**